YarpJS/include/YarpJS_BufferedPort.h

```diff
@@ -31,6 +31,8 @@ class YarpJS_BufferedPort : public YarpJS,
 
     virtual void onRead(T &_datum)
     {
+        onReadCallback->lockMutex();
+
```

Comment:
From a design perspective, I think it would be better to avoid locking the mutex through a public method as I did here, this call not being "tied" to a call to `mutex_callback.unlock()` in any way.
In the old design, `mutex_callback.lock()` was done in `YarpJS_Callback<T>::callCallback()` which schedules the execution of `YarpJS_Callback<T>::_internal_worker(...)` and `YarpJS_Callback<T>::_internal_worker_end(...)` through the function call `uv_async_send( &(this->async) )`. `YarpJS_Callback<T>::_internal_worker_end(...)` calls the `mutex_callback.unlock()`.

For this reason I think it would be better to lock the mutex then latch the `_datum` content (copy to `this->datum`), **both** from `YarpJS_Callback<T>::callCallback()`.

Reply:
I'm pushing an alternative solution...
@traversaro , who else could we assign as reviewer for this kind of change?

Reply:
> @traversaro , who else could we assign as reviewer for this kind of change? @Nicogene ?

Good idea!

Reply:
I never worked with/on yarp.js and I know very few about javascript, sorry :(